services/rollout-service/pkg/datadogdora/datadogdora.go

```diff
@@ -0,0 +1,205 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the Expat(MIT) License as published by
+the Free Software Foundation.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+MIT License for more details.
+
+You should have received a copy of the MIT License
+along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.
+
+Copyright freiheit.com*/
+
+package datadogdora
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadog"
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/freiheit-com/kuberpult/services/rollout-service/pkg/service"
+	"golang.org/x/sync/errgroup"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+type Config struct {
+	URL         string
+	APIKey      string
+	Concurrency int
+	MaxEventAge time.Duration
+}
+
+func New(config Config) *Subscriber {
+	sub := &Subscriber{
+		url:           "",
+		ready:         nil,
+		state:         nil,
+		maxAge:        0,
+		now:           nil,
+		group:         errgroup.Group{},
+		apiKey:        "",
+		doraAPI:       nil,
+		RepositoryUrl: "",
+	}
+	sub.group.SetLimit(config.Concurrency)
+	sub.url = config.URL
+	sub.apiKey = config.APIKey
+	sub.ready = func() {}
+	sub.maxAge = config.MaxEventAge
+	sub.now = time.Now
```

Comment:
(sort of nitpick)
Why not simplify this code like this:
```suggestion
	sub := &Subscriber{
		url:           config.URL,
		ready:         nil,
		state:         nil,
		maxAge:        0,
		now:           nil,
		group:         errgroup.Group{},
		apiKey:        "",
		doraAPI:       nil,
		RepositoryUrl: "",
	}
	sub.group.SetLimit(config.Concurrency)
//	sub.url = config.URL // remove this
	sub.apiKey = config.APIKey
	sub.ready = func() {}
	sub.maxAge = config.MaxEventAge
	sub.now = time.Now
```